homeassistant/components/mqtt/mixins.py

```diff
@@ -868,7 +872,37 @@ async def async_removed_from_registry(self) -> None:
     @callback
     def add_to_platform_abort(self) -> None:
         """Abort adding an entity to a platform."""
+        hook_register: dict[tuple, Callable] = self.hass.data[
+            DATA_MQTT_DISCOVERY_REGISTRY_HOOKS
+        ]
+
+        async def _async_clear_discovery_topic_if_entity_removed(
+            discovery_data: dict[str, Any],
+            event: Event,
+        ) -> None:
+            """Clear the discovery topic when the entiy is removed."""
+            if event.data["action"] == "remove":
+                # publish empty payload to config topic to avoid re-adding
+                hass = async_get_hass()
+                discovery_hash: tuple[str, str] = discovery_data[ATTR_DISCOVERY_HASH]
+                discovery_prefix: str = hass.data[DATA_MQTT].conf[CONF_DISCOVERY_PREFIX]
+                topic = f"{discovery_prefix}/{discovery_hash[0]}/{discovery_hash[1].replace(' ','/')}/config"
```

Comment:
Why do we need to do this, can't we use what's in `discovery_data[ATTR_DISCOVERY_TOPIC]`?